sui-execution/latest/sui-adapter/src/programmable_transactions/trace_utils.rs

```diff
@@ -171,38 +209,218 @@ pub fn trace_split_coins(
+pub fn trace_merge_coins(
+    context: &mut ExecutionContext<'_, '_, '_>,
+    trace_builder_opt: &mut Option<MoveTraceBuilder>,
+    coin_type: &Type,
+    input_infos: &[(u64, ObjectID)],
+    target_coin: &Coin,
+) -> Result<(), ExecutionError> {
+    if let Some(trace_builder) = trace_builder_opt {
+        let type_tag_with_refs = trace_type_to_type_tag_with_refs(context, coin_type)?;
+        let mut input_coin_move_values = vec![];
+        let mut to_merge = 0;
+        for (balance, id) in input_infos {
+            input_coin_move_values.push(coin_move_value_info(
+                type_tag_with_refs.clone(),
+                *id,
+                *balance,
+            )?);
+            to_merge += balance;
+        }
+        let merge_target = coin_move_value_info(
+            type_tag_with_refs.clone(),
+            *target_coin.id.object_id(),
+            target_coin.value() - to_merge,
+        )?;
+        let mut values = vec![ExtMoveValue::Single {
+            name: "merge_target".to_string(),
+            info: merge_target,
+        }];
+        for (idx, input_value) in input_coin_move_values.into_iter().enumerate() {
+            values.push(ExtMoveValue::Single {
+                name: format!("coin{idx}"),
+                info: input_value,
+            });
+        }
+        let merge_result = coin_move_value_info(
+            type_tag_with_refs.clone(),
+            *target_coin.id.object_id(),
+            target_coin.value(),
+        )?;
+        values.push(ExtMoveValue::Single {
+            name: "merge_result".to_string(),
+            info: merge_result,
+        });
+        trace_builder.push_event(TraceEvent::External(Box::new(serde_json::json!(
+            PTBEvent::ExternalEvent(ExternalEvent {
+                description: "MergeCoins: merge_target, coin0...coinN => mergeresult".to_string(),
+                name: "MergeCoins".to_string(),
+                values,
+            })
+        ))));
+    }
+    Ok(())
+}
+
+/// Inserts make move vec event into the trace. As is the case for all other public functions in this module,
+/// its body is (and must be) enclosed in an if statement checking if tracing is enabled.
+pub fn trace_make_move_vec(
+    context: &mut ExecutionContext<'_, '_, '_>,
+    trace_builder_opt: &mut Option<MoveTraceBuilder>,
+    move_values: Vec<ExtMoveValueInfo>,
+    type_: &Type,
+) -> Result<(), ExecutionError> {
+    if let Some(trace_builder) = trace_builder_opt {
+        let type_tag_with_refs = trace_type_to_type_tag_with_refs(context, type_)?;
+        trace_builder.push_event(TraceEvent::External(Box::new(serde_json::json!(
+            PTBEvent::ExternalEvent(ExternalEvent {
+                description: "MakeMoveVec: vec".to_string(),
+                name: "MakeMoveVec".to_string(),
+                values: vec![ExtMoveValue::Vector {
+                    name: "vec".to_string(),
```

Comment:
nit: if I'm understanding what you're going for here correctly, `"vector"` may be better here as that is what is used within Move for making vector values (e.g., `let x = vector[1,2,3];`)